GeneratorInterface/SherpaInterface/src/SherpackUtilities.cc

```diff
@@ -153,6 +154,13 @@ void zerr(int ret)
 /* compress or decompress from stdin to stdout */
 int Unzip(std::string infile, std::string outfile)
 {
+    std::string tmpdir = getenv("TMPDIR");
+    if (tmpdir.size() > 50 ){
+      std::string command = "ln -s $PWD "+tmpdir+"/tmp;";
+      system(command.c_str());
```

Comment:
This does not check if system returns a failure value.

Reply:
A few things to consider in addition to Chris's comment:

- What happens if this path already exists (but is owned by someone else)?
- What happens if `$TMPDIR` is not unique to the job (several jobs run in the same directory, potentially owned by different users)?
- What happens if `$PWD` contains spaces or special characters in it?
- Who cleans up this symlink when the job finishes?

As written, if there's a shared `$TMPDIR` with a long name, one job will run successfully and then all subsequent ones on the worker node will fail.

Reply:
@Dr15Jones @bbockelm investigating a bit further I came across this
https://www.open-mpi.org/faq/?category=osx#startup-errors-with-open-mpi-2.0.x
and the conclusion is:
`The workaround for the Open MPI 2.0.x and v2.1.x release series is to set the TMPDIR environment variable to /tmp or other short directory name.`
as far as I see from the example, the directory created in the `TMPDIR` path should be unique and related to the MPI session.
Would it be an acceptable solution to assing `/tmp/` or `/tmp/$USER/` as `TMPDIR` value in `SherpackUtilities.cc`? I don't see how we could move forward otherwise. Thanks a lot

Reply:
most important, the written data is cleaned up by MPI after use (i.e. if I point `TMPDIR` to any directory, it is empty after the job has either succeeded or failed